cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSClientUtil.java

```diff
@@ -167,4 +179,47 @@ public static void checkFailure(Throwable failure)
         }
     }
 
+    public static void checkTransferFailure(ClientTransfer clientTransfer)
+        throws ResourceNotFoundException, IOException {
+        // check to see if anything went wrong
+        ErrorSummary errorSummary = null;
+
+        try {
+            errorSummary = clientTransfer.getServerError();
+        } catch (Exception e) {
+            log.error("error reading job", e);
+            throw new RuntimeException("Internal error, error reading job:", e);
+        }
+
+        // Check uws job results
+        if (errorSummary != null) {
+            String errorMsg = errorSummary.getSummaryMessage();
+
+            if (StringUtil.hasLength(errorMsg)) {
+                if (errorMsg.contains("Invalid Argument")
+                    || VOS.IVOA_FAULT_INVALID_ARG.equals(errorMsg)) {
+                    log.debug("Invalid Argument found: " + errorMsg);
+                    throw new IllegalArgumentException(VOS.IVOA_FAULT_INVALID_ARG);
+                }
+
+                // not found
+                if (VOS.IVOA_FAULT_NODE_NOT_FOUND.equals(errorMsg) ||
+                    errorMsg.contains("not a data node")) {
```

Comment:
Do you still need to check for the non-standard string messages, like "Invalid Argument" and "not a data node"? 

I've seen "not a data node" thrown quite often so perhaps you do.  This is part of a transfer negotiation step I believe, and is thrown when someone is trying to download a directory (container node) and not a file (data node).  So, it should probably be mapped to IllegalArgumentException instead.

Reply:
I've also seen 'not a data node' thrown in the past few weeks of testing. And items like 'invalidArgument' (no space) - there's only so much testing for non-standard that can be done.

Reply:
Changed 'node not found' to 'IllegalArgumentException'

Reply:
node not found -> ResourceNotFoundException
not a data node -> IllegalArgumentException